util/solc.ts

```diff
@@ -0,0 +1,99 @@
+import { SolidityCompilerInput, SoliditySettings } from 'types/contract'
+
+class SolidityCompiler {
+  worker?: Worker
+  callbacks: { [id: string]: (data: any) => void } = {}
+
+  // NOTE: this lazy-load instead of constructor is due to a bug:
+  // "ReferenceError: Worker is not defined"
+  // this lazy-loading seems to fix it
+  init() {
+    if (!this.worker) {
+      // console.info('starting solc worker')
+      this.worker = new Worker('/solcWorker.js')

```

Comment:
Why is this file used ? 

You added `solcworker2` so I assumed it would be used instead of the old `solcWorker.js`

Reply:
I compile solcWorker2.js to solcWorker.js. I know it's dumb and confusing, but you used the same filename so I wanted to keep it, but also not override your original work. compatibility.